cajita/src/Cajita_SparseArray.hpp

```diff
@@ -0,0 +1,541 @@
+#include <Cajita_MpiTraits.hpp>
+#include <Cajita_SparseIndexSpace.hpp>
+#include <Cajita_Types.hpp>
+
+#include <Cabana_AoSoA.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <cmath>
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+#include <mpi.h>
+
+namespace Cajita
+{
+
+namespace Experimental
+{
+//---------------------------------------------------------------------------//
+// Indexing type tags
+//---------------------------------------------------------------------------//
+// H - Hierachical
+// T_C: indexing manner: tile ijk - cell ijk
+struct Index_H_T_C
```

Comment:
From yesterday, let's see if it's possible to differentiate overloads without these tags (using `Kokkos::Array` or `std::array`